tests/model/test_where_filter_spec.py

```diff
@@ -352,6 +360,34 @@ def test_entity_in_filter(  # noqa: D103
         dimension_specs=(),
         time_dimension_specs=(),
         entity_specs=(EntitySpec(element_name="user", entity_links=(EntityReference(element_name="listing"),)),),
+        group_by_metric_specs=(),
+    )
+
+
+def test_metric_in_filter(  # noqa: D103
+    column_association_resolver: ColumnAssociationResolver,
+    resolved_spec_lookup: FilterSpecResolutionLookUp,
+) -> None:
+    where_filter = PydanticWhereFilter(where_sql_template="{{ Metric('bookings', group_by=['listing']) }} > 2")
+
+    group_by_metric_spec = GroupByMetricSpec(element_name="bookings", entity_links=(EntityReference("listing"),))
```

Comment:
Yeah I should have left a comment about this because I was aware of it. But figured at this point, it essentially IS an entity link since we only allow one entity and that's what we join on.
But this is definitely part of what will make it complicated to add dimensions 😅

I do think we will need to remove `entity_links` from the `GroupByMetricSpec` but TBD how extensive the logic changes will be to deal with that...